File: packages/barcode-scanning/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/barcodescanning/classes/results/GetMaxZoomRatioResult.java
package io.capawesome.capacitorjs.plugins.mlkit.barcodescanning.classes.results;

import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.mlkit.barcodescanning.interfaces.Result;

public class GetMaxZoomRatioResult implements Result {

    private float zoomRatio;

    public GetMaxZoomRatioResult(float zoomRatio) {
        this.zoomRatio = zoomRatio;
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("zoomRatio", zoomRatio);
        return result;
    }
}

File: packages/barcode-scanning/ios/Plugin/Classes/Results/GetMaxZoomRatioResult.swift
import Foundation
import Capacitor

@objc public class GetMaxZoomRatioResult: NSObject, Result {
    let zoomRatio: Float

    init(zoomRatio: CGFloat) {
        self.zoomRatio = Float(zoomRatio)
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["zoomRatio"] = zoomRatio
        return result as AnyObject
    }
}
